exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/QueryResources.java

```diff
@@ -110,6 +118,50 @@ public Viewable submitQuery(@FormParam("query") String query,
     }
   }
 
+  private Map<String, Object> readOptionsFromForm(Form form) {
+    Map<String, Object> options = new HashMap<>();
+    SessionOptionManager sessionOptionManager = webUserConnection.getSession().getOptions();
+    for (Map.Entry<String, List<String>> pair : form.asMap().entrySet()) {
+      String name = pair.getKey();
+      OptionDefinition definition = sessionOptionManager.getOptionDefinition(name);
+      if (definition != null) {
+        String valueStr = pair.getValue().get(0);
+        if (!valueStr.isEmpty()) {
+          try {
+            Object value = null;
+            switch (definition.getValidator().getKind()) {
+              case BOOLEAN:
+                if ("true".equals(valueStr)) {
```

Comment:
What is user passes true in upper case? Camel case? I still think `Boolean.valueOf()` is the most optimal solution though...

Reply:
`Boolean.valueOf` doesn't handle any of the cases you mentioned, and doesn't validate that the input is actually either `true` or `false`.  I only wanted to accept valid JSON/Java compatible boolean values and alert the user if they provided an invalid value.

Reply:
We should mind current Drill behavior in this case. Currently user can set boolean option using in any case (current example for `true` value, the same is for false, other values are not allowed):
```
set metastore.enabled = TRUE;
set metastore.enabled = True;
set metastore.enabled = true;
```
I think for the user it would be a surprise that behavior is different. So I think we should add case-insensitivity for boolean options.

Reply:
@dobesv I know you are not interested in proper implementation but frankly saying much better solution was to mimic the way options are updated on Options tab, user would under option name, its kind (from drop down) and depending on the kind value. And then update process would require only several line in the server side. Please see example: https://github.com/apache/drill/blob/ab60b3d8e5d062edaccb9d292fac95d076d322e5/exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/StatusResources.java#L183